### PR TITLE
Added `ProgressBar` component

### LINKED ISSUE #2172: Build the `Progress` component
Refer the [ProgressBar](https://github.com/bigbinary/neeto-form-web/blob/e0e544b632d387a32ae5463084c2e3ebc4a99943/app/javascript/src/components/commons/ProgressBar.jsx) component from neetoForm.

- `progressValue` should be optional. If not provided, it should be equal to `progressPercentage%`.
---

**Rollout**

- [ ] neetoForm - export submissions
- [ ] neetoQuiz - export submissions
- [ ] neeto-image-uploader-nano
- [ ] neeto-editor
- [ ] -- check for file upload implementations --

---

engineer-needed _t

### PR DESCRIPTION
**Pull request by gaagul**
- Fixes #2172 

**Description**

- Added: a new component `ProgressBar`

**Checklist**

- [x] I have made corresponding changes to the documentation.
- [x] I have updated the types definition of modified exports.
- [x] I have verified the functionality in some of the neeto web-apps.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added proper `data-cy` and `data-testid` attributes.
- [x] I have added the necessary label (`patch`/`minor`/`major` - If package publish
      is required).

**Reviewers**

@gaagul _a

### CODE REVIEW

**Comment by gaagul** [comment]
@josephmathew900 _A please review.

File: tests/ProgressBar.test.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+import ProgressBar from "components/ProgressBar";
+
+describe("ProgressBar", () => {
+  it("renders with correct value", () => {
+    render(<ProgressBar progressPercentage={50} progressValue="50%" />);
+
+    expect(screen.getByText("50%")).toBeInTheDocument();
+  });
+});
```
Comment (josephmathew900) [inline tests/ProgressBar.test.jsx:13]: Add a test for - `progressValue should be optional. If not provided, it should be equal to progressPercentage%`.

**Review by josephmathew900 — APPROVED** [review]
@gaagul _a Added a minor suggestion. Rest LGTM. After making the change please assign the PR to Praveen for UI review.

**Comment by gaagul** [comment]
@praveen-murali-ind _a Please review the UI.

**Review by praveen-murali-ind — APPROVED** [review]
LGTM. 🎉

**Pull request merged**